wgpu-core/src/command/bind.rs

```diff
@@ -213,3 +213,54 @@ impl Binder {
             .unwrap_or_else(|| self.entries.len())
     }
 }
+
+struct PushConstantChange {
+    stages: wgt::ShaderStage,
+    offset: u32,
+    enable: bool,
+}
+
+/// Break up possibly overlapping push constant ranges into a set of non-overlapping ranges
+/// which contain all the stage flags of the original ranges. This allows us to zero out (or write any value)
+/// to every possible value.
+pub fn compute_nonoverlapping_ranges(
+    ranges: &[wgt::PushConstantRange],
+) -> ArrayVec<[wgt::PushConstantRange; crate::device::SHADER_STAGE_COUNT * 2]> {
```

Comment:
we don't generally use fully qualified paths to our own types in wgpu. Could we have that included in the `use crate::{..}` at the top?